src/init.cpp

```diff
@@ -598,6 +602,19 @@ bool AppInit2(boost::thread_group& threadGroup)
     if (nFD - MIN_CORE_FILEDESCRIPTORS < nMaxConnections)
         nMaxConnections = nFD - MIN_CORE_FILEDESCRIPTORS;
 
+    if (GetArg("-prune", 0)) {
+        if (GetBoolArg("-txindex", false))
+            return InitError(_("Prune mode is incompatible with -txindex."));
+#ifdef ENABLE_WALLET
+        if (!GetBoolArg("-disablewallet", false)) {
+            if (SoftSetBoolArg("-disablewallet", true))
+                LogPrintf("%s : parameter interaction: -prune -> setting -disablewallet=1\n", __func__);
```

Comment:
I was wrong in my above comment, it is possible to use the wallet with pruning as of #6057. It just excludes some operations such as rescanning which need the entire chain.

SPV mode is orthogonal to that, though. That has always been desirable, and pruning hasn't changed that.